ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -790,11 +790,12 @@ private void prepareForSending() {
         chunkBlobId =
             new BlobId(BlobId.DEFAULT_FLAG, clusterMap.getLocalDatacenterId(), passedInBlobProperties.getAccountId(),
                 passedInBlobProperties.getContainerId(), partitionId);
+        // @todo: set encrypted flag as appropriate.
         chunkBlobProperties = new BlobProperties(buf.remaining(), passedInBlobProperties.getServiceId(),
             passedInBlobProperties.getOwnerId(), passedInBlobProperties.getContentType(),
             passedInBlobProperties.isPrivate(), passedInBlobProperties.getTimeToLiveInSeconds(),
             passedInBlobProperties.getCreationTimeInMs(), passedInBlobProperties.getAccountId(),
-            passedInBlobProperties.getContainerId());
+            passedInBlobProperties.getContainerId(), false);
```

Comment:
BlobProperties already has isEncrypted() which is added in this patch. You can call into it.

Reply:
But will it be part of the `passedInBlobProperties`? Who decides to add the flag?

Reply:
yes, RestUtils.buildBlobProperties(....) will construct BlobProps with encryption flag based on container fields.